oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/profiling/ebpf/analyze/EBPFProfilingAnalyzer.java

```diff
@@ -0,0 +1,188 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.oap.server.core.profiling.ebpf.analyze;
+
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.skywalking.oap.server.core.profiling.ebpf.storage.EBPFProfilingDataRecord;
+import org.apache.skywalking.oap.server.core.query.type.EBPFProfilingAnalyzation;
+import org.apache.skywalking.oap.server.core.query.type.EBPFProfilingAnalyzeTimeRange;
+import org.apache.skywalking.oap.server.core.query.type.EBPFProfilingTree;
+import org.apache.skywalking.oap.server.core.storage.StorageModule;
+import org.apache.skywalking.oap.server.core.storage.profiling.ebpf.IEBPFProfilingDataDAO;
+import org.apache.skywalking.oap.server.library.module.ModuleManager;
+import org.apache.skywalking.oap.server.library.util.CollectionUtils;
+import org.apache.skywalking.oap.server.library.util.StringUtil;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * eBPF Profiling Analyzer working on data query and combine them for generate the Flame Graph.
+ */
+@Slf4j
+public class EBPFProfilingAnalyzer {
+
+    private static final EBPFProfilingAnalyzeCollector ANALYZE_COLLECTOR = new EBPFProfilingAnalyzeCollector();
+    private static final Long FETCH_DATA_DURATION = TimeUnit.MINUTES.toMillis(2);
+    private static final ExecutorService FETCH_DATA_THREAD_POOL = Executors.newCachedThreadPool();
```

Comment:
The maximumPoolSize of the thread pool should be fixed and configurable. Default, it is Int.MAX, which could easily crash OAP server. We need to reject or timeout when face too many analysis requests

Reply:
This thread pool size is the same as the java ForkJoin pool size, should I update to `Runtime.getRuntime().availableProcessors()`?

Reply:
```java
    public static ExecutorService newCachedThreadPool() {
        return new ThreadPoolExecutor(0, Integer.MAX_VALUE,
                                      60L, TimeUnit.SECONDS,
                                      new SynchronousQueue<Runnable>());
    }
```

From my local source codes, the 2nd parameter is `maximumPoolSize`, which is `Integer.MAX_VALUE`.

Reply:
`Runtime.getRuntime().availableProcessors()` could be default, but it should be configurable for different use cases. Such as some users may want less CPU working for analysis, more for streaming aggregation process.

Reply:
Just say, using `Runtime.getRuntime().availableProcessors()` in every time we need a thread pool, globally, the number of threads would also be crazy :) Isn't it.

Reply:
Yes, may we need two global thread pools for doing that. One for the profiling data analysis, the other one for the wait for the data query(works for the circle break the query when timeout). WDYT?

Reply:
Let's consider that later. For now, let's just make sure `Runtime.getRuntime().availableProcessors()` as default and support configuration through application.yml

Reply:
Added.